--- binutils-2.22/binutils/readelf.c.~1~	2014-03-01 10:38:01.000000000 +0900
+++ binutils-2.22/binutils/readelf.c	2014-03-01 11:07:08.000000000 +0900
@@ -8878,7 +8878,7 @@
 
   n = print_vma (si, DEC_5);
   if (n < 5)
-    fputs ("     " + n, stdout);
+    fputs (&"     "[n], stdout);
   printf (" %3lu: ", hn);
   print_vma (psym->st_value, LONG_HEX);
   putchar (' ');
--- binutils-2.22/ld/ldmisc.c.~1~	2011-05-23 15:13:35.000000000 +0900
+++ binutils-2.22/ld/ldmisc.c	2014-03-01 11:09:55.000000000 +0900
@@ -351,7 +351,7 @@
 		    if (functionname != NULL && fmt[-1] == 'G')
 		      lfinfo (fp, "%T", functionname);
 		    else if (filename != NULL && linenumber != 0)
-		      fprintf (fp, "%u%s", linenumber, ":" + done);
+		      fprintf (fp, "%u%s", linenumber, &":"[done]);
 		    else
 		      done = FALSE;
 		  }
