--- binutils-2.20.1-dist/gas/config/tc-msp430.h	2007-07-03 20:01:04.000000000 +0900
+++ binutils-2.20.1/gas/config/tc-msp430.h	2010-06-12 23:22:20.000000000 +0900
@@ -99,6 +99,7 @@
 
 #define LEX_DOLLAR 0
 /* MSP430 port does not use `$' as a logical line separator */
+#undef LEX_DOLLAR
 
 #define TC_IMPLICIT_LCOMM_ALIGNMENT(SIZE, P2VAR) (P2VAR) = 0
 /*   An `.lcomm' directive with no explicit alignment parameter will
