diff --git a/gcc/config/msp430/generate_chip_data.py b/gcc/config/msp430/generate_chip_data.py
index f55f7b1..5d4fa64 100644
--- a/gcc/config/msp430/generate_chip_data.py
+++ b/gcc/config/msp430/generate_chip_data.py
@@ -84,10 +84,36 @@ text = '''
 	{"msp430x2122",  MSP430_ISA_22, 0},
 	{"msp430x2132",  MSP430_ISA_22, 0},
 
-	{"msp430x2201",  MSP430_ISA_22, 0}, /* Value-Line */
-	{"msp430x2211",  MSP430_ISA_22, 0}, /* Value-Line */
-	{"msp430x2221",  MSP430_ISA_22, 0}, /* Value-Line */
-	{"msp430x2231",  MSP430_ISA_22, 0}, /* Value-Line */
+	{"msp430g2001",  MSP430_ISA_22, 0},
+	{"msp430g2101",  MSP430_ISA_22, 0},
+	{"msp430g2111",  MSP430_ISA_22, 0},
+	{"msp430g2201",  MSP430_ISA_22, 0},
+	{"msp430g2211",  MSP430_ISA_22, 0},
+
+	{"msp430g2121",  MSP430_ISA_22, 0},
+	{"msp430g2131",  MSP430_ISA_22, 0},
+	{"msp430g2221",  MSP430_ISA_22, 0},
+	{"msp430g2231",  MSP430_ISA_22, 0},
+
+	{"msp430g2102",  MSP430_ISA_22, 0},
+	{"msp430g2202",  MSP430_ISA_22, 0},
+	{"msp430g2302",  MSP430_ISA_22, 0},
+	{"msp430g2402",  MSP430_ISA_22, 0},
+
+	{"msp430g2132",  MSP430_ISA_22, 0},
+	{"msp430g2232",  MSP430_ISA_22, 0},
+	{"msp430g2332",  MSP430_ISA_22, 0},
+	{"msp430g2432",  MSP430_ISA_22, 0},
+
+	{"msp430g2112",  MSP430_ISA_22, 0},
+	{"msp430g2212",  MSP430_ISA_22, 0},
+	{"msp430g2312",  MSP430_ISA_22, 0},
+	{"msp430g2412",  MSP430_ISA_22, 0},
+
+	{"msp430g2152",  MSP430_ISA_22, 0},
+	{"msp430g2252",  MSP430_ISA_22, 0},
+	{"msp430g2352",  MSP430_ISA_22, 0},
+	{"msp430g2452",  MSP430_ISA_22, 0},
 
 	{"msp430x2232",  MSP430_ISA_22, 0},
 	{"msp430x2252",  MSP430_ISA_22, 0},
@@ -298,7 +324,7 @@ for line in text.splitlines():
     if (exemplary_chip[mlib] == chip):
         aux_cbs = '|mmcu=%s' % (mlib,)
     generated.setdefault('MULTILIB_MATCHES', []).append("\tmmcu?%s=mmcu?%s" % (mlib, chip))
-    generated.setdefault('CPP_SPEC', []).append('%%{mmcu=%s:%%(cpp_%s) -D__%s__}' % (chip, mlib, chip.replace('x', '_').upper()))
+    generated.setdefault('CPP_SPEC', []).append('%%{mmcu=%s:%%(cpp_%s) -D__%s__}' % (chip, mlib, chip.replace('msp430g2', 'msp430_g2').replace('x', '_').upper()))
     generated.setdefault('LINK_SPEC', []).append('%%{mmcu=%s:-m %s }' % (chip, chip))
     generated.setdefault('CRT_BINUTILS_SPECS', []).append('%%{mmcu=%s%s:crt%s.o%%s}' % (chip, aux_cbs, chip.replace('msp', '')))
     
diff --git a/gcc/config/msp430/msp430.c b/gcc/config/msp430/msp430.c
index 9fdc0c9..fd295a8 100644
--- a/gcc/config/msp430/msp430.c
+++ b/gcc/config/msp430/msp430.c
@@ -190,10 +190,36 @@ static struct mcu_type_s msp430_mcu_types[] = {
 	{"msp430x2122",  MSP430_ISA_22, 0},
 	{"msp430x2132",  MSP430_ISA_22, 0},
 
-	{"msp430x2201",  MSP430_ISA_22, 0}, /* Value-Line */
-	{"msp430x2211",  MSP430_ISA_22, 0}, /* Value-Line */
-	{"msp430x2221",  MSP430_ISA_22, 0}, /* Value-Line */
-	{"msp430x2231",  MSP430_ISA_22, 0}, /* Value-Line */
+	{"msp430g2001",  MSP430_ISA_22, 0},
+	{"msp430g2101",  MSP430_ISA_22, 0},
+	{"msp430g2111",  MSP430_ISA_22, 0},
+	{"msp430g2201",  MSP430_ISA_22, 0},
+	{"msp430g2211",  MSP430_ISA_22, 0},
+
+	{"msp430g2121",  MSP430_ISA_22, 0},
+	{"msp430g2131",  MSP430_ISA_22, 0},
+	{"msp430g2221",  MSP430_ISA_22, 0},
+	{"msp430g2231",  MSP430_ISA_22, 0},
+
+	{"msp430g2102",  MSP430_ISA_22, 0},
+	{"msp430g2202",  MSP430_ISA_22, 0},
+	{"msp430g2302",  MSP430_ISA_22, 0},
+	{"msp430g2402",  MSP430_ISA_22, 0},
+
+	{"msp430g2132",  MSP430_ISA_22, 0},
+	{"msp430g2232",  MSP430_ISA_22, 0},
+	{"msp430g2332",  MSP430_ISA_22, 0},
+	{"msp430g2432",  MSP430_ISA_22, 0},
+
+	{"msp430g2112",  MSP430_ISA_22, 0},
+	{"msp430g2212",  MSP430_ISA_22, 0},
+	{"msp430g2312",  MSP430_ISA_22, 0},
+	{"msp430g2412",  MSP430_ISA_22, 0},
+
+	{"msp430g2152",  MSP430_ISA_22, 0},
+	{"msp430g2252",  MSP430_ISA_22, 0},
+	{"msp430g2352",  MSP430_ISA_22, 0},
+	{"msp430g2452",  MSP430_ISA_22, 0},
 
 	{"msp430x2232",  MSP430_ISA_22, 0},
 	{"msp430x2252",  MSP430_ISA_22, 0},
diff --git a/gcc/config/msp430/msp430.h b/gcc/config/msp430/msp430.h
index b60d674..dea8225 100644
--- a/gcc/config/msp430/msp430.h
+++ b/gcc/config/msp430/msp430.h
@@ -2565,10 +2565,31 @@ valid_machine_decl_attribute (DECL, ATTRIBUTES, IDENTIFIER, ARGS)
 %{mmcu=msp430x2112:%(cpp_msp1) -D__MSP430_2112__} \
 %{mmcu=msp430x2122:%(cpp_msp1) -D__MSP430_2122__} \
 %{mmcu=msp430x2132:%(cpp_msp1) -D__MSP430_2132__} \
-%{mmcu=msp430x2201:%(cpp_msp1) -D__MSP430_2201__} \
-%{mmcu=msp430x2211:%(cpp_msp1) -D__MSP430_2211__} \
-%{mmcu=msp430x2221:%(cpp_msp1) -D__MSP430_2221__} \
-%{mmcu=msp430x2231:%(cpp_msp1) -D__MSP430_2231__} \
+%{mmcu=msp430g2001:%(cpp_msp1) -D__MSP430_G2001__} \
+%{mmcu=msp430g2101:%(cpp_msp1) -D__MSP430_G2101__} \
+%{mmcu=msp430g2111:%(cpp_msp1) -D__MSP430_G2111__} \
+%{mmcu=msp430g2201:%(cpp_msp1) -D__MSP430_G2201__} \
+%{mmcu=msp430g2211:%(cpp_msp1) -D__MSP430_G2211__} \
+%{mmcu=msp430g2121:%(cpp_msp1) -D__MSP430_G2121__} \
+%{mmcu=msp430g2131:%(cpp_msp1) -D__MSP430_G2131__} \
+%{mmcu=msp430g2221:%(cpp_msp1) -D__MSP430_G2221__} \
+%{mmcu=msp430g2231:%(cpp_msp1) -D__MSP430_G2231__} \
+%{mmcu=msp430g2102:%(cpp_msp1) -D__MSP430_G2102__} \
+%{mmcu=msp430g2202:%(cpp_msp1) -D__MSP430_G2202__} \
+%{mmcu=msp430g2302:%(cpp_msp1) -D__MSP430_G2302__} \
+%{mmcu=msp430g2402:%(cpp_msp1) -D__MSP430_G2402__} \
+%{mmcu=msp430g2132:%(cpp_msp1) -D__MSP430_G2132__} \
+%{mmcu=msp430g2232:%(cpp_msp1) -D__MSP430_G2232__} \
+%{mmcu=msp430g2332:%(cpp_msp1) -D__MSP430_G2332__} \
+%{mmcu=msp430g2432:%(cpp_msp1) -D__MSP430_G2432__} \
+%{mmcu=msp430g2112:%(cpp_msp1) -D__MSP430_G2112__} \
+%{mmcu=msp430g2212:%(cpp_msp1) -D__MSP430_G2212__} \
+%{mmcu=msp430g2312:%(cpp_msp1) -D__MSP430_G2312__} \
+%{mmcu=msp430g2412:%(cpp_msp1) -D__MSP430_G2412__} \
+%{mmcu=msp430g2152:%(cpp_msp1) -D__MSP430_G2152__} \
+%{mmcu=msp430g2252:%(cpp_msp1) -D__MSP430_G2252__} \
+%{mmcu=msp430g2352:%(cpp_msp1) -D__MSP430_G2352__} \
+%{mmcu=msp430g2452:%(cpp_msp1) -D__MSP430_G2452__} \
 %{mmcu=msp430x2232:%(cpp_msp1) -D__MSP430_2232__} \
 %{mmcu=msp430x2252:%(cpp_msp1) -D__MSP430_2252__} \
 %{mmcu=msp430x2272:%(cpp_msp1) -D__MSP430_2272__} \
@@ -2783,10 +2804,31 @@ valid_machine_decl_attribute (DECL, ATTRIBUTES, IDENTIFIER, ARGS)
 %{mmcu=msp430x2112:-m msp430x2112 } \
 %{mmcu=msp430x2122:-m msp430x2122 } \
 %{mmcu=msp430x2132:-m msp430x2132 } \
-%{mmcu=msp430x2201:-m msp430x2201 } \
-%{mmcu=msp430x2211:-m msp430x2211 } \
-%{mmcu=msp430x2221:-m msp430x2221 } \
-%{mmcu=msp430x2231:-m msp430x2231 } \
+%{mmcu=msp430g2001:-m msp430g2001 } \
+%{mmcu=msp430g2101:-m msp430g2101 } \
+%{mmcu=msp430g2111:-m msp430g2111 } \
+%{mmcu=msp430g2201:-m msp430g2201 } \
+%{mmcu=msp430g2211:-m msp430g2211 } \
+%{mmcu=msp430g2121:-m msp430g2121 } \
+%{mmcu=msp430g2131:-m msp430g2131 } \
+%{mmcu=msp430g2221:-m msp430g2221 } \
+%{mmcu=msp430g2231:-m msp430g2231 } \
+%{mmcu=msp430g2102:-m msp430g2102 } \
+%{mmcu=msp430g2202:-m msp430g2202 } \
+%{mmcu=msp430g2302:-m msp430g2302 } \
+%{mmcu=msp430g2402:-m msp430g2402 } \
+%{mmcu=msp430g2132:-m msp430g2132 } \
+%{mmcu=msp430g2232:-m msp430g2232 } \
+%{mmcu=msp430g2332:-m msp430g2332 } \
+%{mmcu=msp430g2432:-m msp430g2432 } \
+%{mmcu=msp430g2112:-m msp430g2112 } \
+%{mmcu=msp430g2212:-m msp430g2212 } \
+%{mmcu=msp430g2312:-m msp430g2312 } \
+%{mmcu=msp430g2412:-m msp430g2412 } \
+%{mmcu=msp430g2152:-m msp430g2152 } \
+%{mmcu=msp430g2252:-m msp430g2252 } \
+%{mmcu=msp430g2352:-m msp430g2352 } \
+%{mmcu=msp430g2452:-m msp430g2452 } \
 %{mmcu=msp430x2232:-m msp430x2232 } \
 %{mmcu=msp430x2252:-m msp430x2252 } \
 %{mmcu=msp430x2272:-m msp430x2272 } \
@@ -2980,10 +3022,31 @@ valid_machine_decl_attribute (DECL, ATTRIBUTES, IDENTIFIER, ARGS)
 %{mmcu=msp430x2112:crt430x2112.o%s} \
 %{mmcu=msp430x2122:crt430x2122.o%s} \
 %{mmcu=msp430x2132:crt430x2132.o%s} \
-%{mmcu=msp430x2201:crt430x2201.o%s} \
-%{mmcu=msp430x2211:crt430x2211.o%s} \
-%{mmcu=msp430x2221:crt430x2221.o%s} \
-%{mmcu=msp430x2231:crt430x2231.o%s} \
+%{mmcu=msp430g2001:crt430g2001.o%s} \
+%{mmcu=msp430g2101:crt430g2101.o%s} \
+%{mmcu=msp430g2111:crt430g2111.o%s} \
+%{mmcu=msp430g2201:crt430g2201.o%s} \
+%{mmcu=msp430g2211:crt430g2211.o%s} \
+%{mmcu=msp430g2121:crt430g2121.o%s} \
+%{mmcu=msp430g2131:crt430g2131.o%s} \
+%{mmcu=msp430g2221:crt430g2221.o%s} \
+%{mmcu=msp430g2231:crt430g2231.o%s} \
+%{mmcu=msp430g2102:crt430g2102.o%s} \
+%{mmcu=msp430g2202:crt430g2202.o%s} \
+%{mmcu=msp430g2302:crt430g2302.o%s} \
+%{mmcu=msp430g2402:crt430g2402.o%s} \
+%{mmcu=msp430g2132:crt430g2132.o%s} \
+%{mmcu=msp430g2232:crt430g2232.o%s} \
+%{mmcu=msp430g2332:crt430g2332.o%s} \
+%{mmcu=msp430g2432:crt430g2432.o%s} \
+%{mmcu=msp430g2112:crt430g2112.o%s} \
+%{mmcu=msp430g2212:crt430g2212.o%s} \
+%{mmcu=msp430g2312:crt430g2312.o%s} \
+%{mmcu=msp430g2412:crt430g2412.o%s} \
+%{mmcu=msp430g2152:crt430g2152.o%s} \
+%{mmcu=msp430g2252:crt430g2252.o%s} \
+%{mmcu=msp430g2352:crt430g2352.o%s} \
+%{mmcu=msp430g2452:crt430g2452.o%s} \
 %{mmcu=msp430x2232:crt430x2232.o%s} \
 %{mmcu=msp430x2252:crt430x2252.o%s} \
 %{mmcu=msp430x2272:crt430x2272.o%s} \
diff --git a/gcc/config/msp430/t-msp430 b/gcc/config/msp430/t-msp430
index 267f9fd..b4bcbc0 100644
--- a/gcc/config/msp430/t-msp430
+++ b/gcc/config/msp430/t-msp430
@@ -108,10 +108,31 @@ MULTILIB_MATCHES = \
 	mmcu?msp1=mmcu?msp430x2112 \
 	mmcu?msp1=mmcu?msp430x2122 \
 	mmcu?msp1=mmcu?msp430x2132 \
-	mmcu?msp1=mmcu?msp430x2201 \
-	mmcu?msp1=mmcu?msp430x2211 \
-	mmcu?msp1=mmcu?msp430x2221 \
-	mmcu?msp1=mmcu?msp430x2231 \
+	mmcu?msp1=mmcu?msp430g2001 \
+	mmcu?msp1=mmcu?msp430g2101 \
+	mmcu?msp1=mmcu?msp430g2111 \
+	mmcu?msp1=mmcu?msp430g2201 \
+	mmcu?msp1=mmcu?msp430g2211 \
+	mmcu?msp1=mmcu?msp430g2121 \
+	mmcu?msp1=mmcu?msp430g2131 \
+	mmcu?msp1=mmcu?msp430g2221 \
+	mmcu?msp1=mmcu?msp430g2231 \
+	mmcu?msp1=mmcu?msp430g2102 \
+	mmcu?msp1=mmcu?msp430g2202 \
+	mmcu?msp1=mmcu?msp430g2302 \
+	mmcu?msp1=mmcu?msp430g2402 \
+	mmcu?msp1=mmcu?msp430g2132 \
+	mmcu?msp1=mmcu?msp430g2232 \
+	mmcu?msp1=mmcu?msp430g2332 \
+	mmcu?msp1=mmcu?msp430g2432 \
+	mmcu?msp1=mmcu?msp430g2112 \
+	mmcu?msp1=mmcu?msp430g2212 \
+	mmcu?msp1=mmcu?msp430g2312 \
+	mmcu?msp1=mmcu?msp430g2412 \
+	mmcu?msp1=mmcu?msp430g2152 \
+	mmcu?msp1=mmcu?msp430g2252 \
+	mmcu?msp1=mmcu?msp430g2352 \
+	mmcu?msp1=mmcu?msp430g2452 \
 	mmcu?msp1=mmcu?msp430x2232 \
 	mmcu?msp1=mmcu?msp430x2252 \
 	mmcu?msp1=mmcu?msp430x2272 \
