diff --git a/scripts/install.sh b/scripts/install.sh
index 3d71e63..6a6768d 100755
--- a/scripts/install.sh
+++ b/scripts/install.sh
@@ -4,7 +4,7 @@
 # This file is part of msp430mcu (http://sourceforge.net/projects/mspgcc/)
 
 PREFIX=${1:-/msp430/install/dev}
-MSP430MCU_ROOT=${MSP430MCU_ROOT:-${MSP430_ROOT:-/msp430}/msp430mcu}
+MSP430MCU_ROOT=${2:-${MSP430MCU_ROOT:-${MSP430_ROOT:-/msp430}/msp430mcu}}
 UPSTREAM=${MSP430MCU_ROOT}/upstream
 ANALYSIS=${MSP430MCU_ROOT}/analysis
 SCRIPTS=${MSP430MCU_ROOT}/scripts
