--- a/src/Makefile	2011-06-13 03:29:42.000000000 +0900
+++ b/src/Makefile	2011-09-04 09:46:28.778910433 +0900
@@ -20,9 +20,9 @@
 SRC_ROOT := .
 BUILD_ROOT = ./Build
 
-override CC = msp430-gcc
-override AS = msp430-gcc -x assembler-with-cpp
-override AR = msp430-ar
+override CC = $(PREFIX)/bin/msp430-gcc
+override AS = $(PREFIX)/bin/msp430-gcc -x assembler-with-cpp
+override AR = $(PREFIX)/bin/msp430-ar
 RM = rm
 MD = mkdir -p $@
 
@@ -69,8 +69,8 @@
 install-multilib:
 	for i in `$(CC) --print-multi-lib 2>/dev/null`; do \
 	  dir=`echo $$i | sed -e 's/;.*$$//'` ; \
-	  mkdir -p $(libdir)/$$dir; \
-	  $(MAKE) $(MFLAGS) MULTILIB_DIR="$$dir" install-libc install-libm ; \
+	  sudo mkdir -p $(libdir)/$$dir; \
+	  sudo $(MAKE) $(MFLAGS) PREFIX=$(PREFIX) MULTILIB_DIR="$$dir" install-libc install-libm ; \
 	done;
 
 .PHONY: clean-multilib clean-libc clean-libm
