--- a/Makefile	2011-11-08 06:11:49.000000000 +0900
+++ b/Makefile	2012-01-08 23:51:58.000000000 +0900
@@ -36,8 +36,9 @@
 
 UNAME := $(shell sh -c 'uname -s')
 ifeq ($(UNAME),Darwin) # Mac OS X/MacPorts stuff
-	PORTS_CFLAGS = -I/opt/local/include
-	PORTS_LDFLAGS = -L/opt/local/lib
+	PORTS_CFLAGS = -I/opt/local/include -I/opt/local/include/libusb-legacy
+	PORTS_LDFLAGS = -L/opt/local/lib -L/opt/local/lib/libusb-legacy
+	LIBUSB_LIBS = -lusb-legacy
 else
   ifeq ($(UNAME),OpenBSD) # OpenBSD Ports stuff
 	PORTS_CFLAGS = `pkg-config --cflags libusb`
@@ -46,6 +47,7 @@
 	PORTS_CFLAGS =
 	PORTS_LDFLAGS =
   endif
+  LIBUSB_LIBS = -lusb
 endif
 
 ifeq ($(OS),Windows_NT)
@@ -70,7 +72,7 @@
 CONFIG_CFLAGS = -DLIB_DIR=\"$(LIBDIR)\"
 
 MSPDEBUG_LDFLAGS = $(LDFLAGS) $(PORTS_LDFLAGS)
-MSPDEBUG_LIBS = -lusb $(READLINE_LIBS) $(OS_LIBS)
+MSPDEBUG_LIBS = $(LIBUSB_LIBS) $(READLINE_LIBS) $(OS_LIBS)
 MSPDEBUG_CFLAGS = $(CFLAGS) $(READLINE_CFLAGS) $(PORTS_CFLAGS)\
  $(GCC_CFLAGS) $(INCLUDES) $(CONFIG_CFLAGS)
 
