diff --git a/uif.c b/uif.c
index 627f98e..8221834 100644
--- a/uif.c
+++ b/uif.c
@@ -25,7 +25,9 @@
 #include <unistd.h>
 #include <termios.h>
 
+#ifndef __APPLE__
 #include <linux/serial.h>
+#endif
 #include <sys/ioctl.h>
 #include <fcntl.h>
 
@@ -89,7 +91,9 @@ static int open_olimex_iso(const char *device)
 {
         int fd = open(device, O_RDWR | O_NOCTTY);
         struct termios attr;
+#ifdef TIOCSSERIAL
         struct serial_struct serial_info;
+#endif
 
         if (fd < 0)
                 return -1;
@@ -99,6 +103,7 @@ static int open_olimex_iso(const char *device)
         cfsetispeed(&attr, B38400);
         cfsetospeed(&attr, B38400);
 
+#ifdef TIOCSSERIAL
         serial_info.flags = ASYNC_SPD_CUST;
         serial_info.custom_divisor = 120;
         if (ioctl(fd, TIOCSSERIAL, &serial_info) < 0) {
@@ -106,6 +111,7 @@ static int open_olimex_iso(const char *device)
 			   strerror(errno));
 		return -1;
         }
+#endif
 
         if (tcsetattr(fd, TCSAFLUSH, &attr) < 0)
                 return -1;
