diff --git a/Makefile b/Makefile
index bd8cd5e..eccd4c7 100644
--- a/Makefile
+++ b/Makefile
@@ -60,7 +60,7 @@ mspdebug: main.o fet.o rf2500.o dis.o uif.o olimex.o ihex.o elf32.o stab.o \
 	  reader.o vector.o output_util.o expr.o fet_error.o binfile.o \
 	  fet_db.o usbutil.o titext.o srec.o device.o coff.o opdb.o output.o \
 	  cmddb.o stdcmd.o prog.o flash_bsl.o
-	$(CC) $(LDFLAGS) $(MACPORTS_LDFLAGS) -o $@ $^ -lusb $(READLINE_LIBS)
+	$(CC) $(LDFLAGS) $(MACPORTS_LDFLAGS) -o $@ $^ $(READLINE_LIBS)
 
 .c.o:
 	$(CC) $(CFLAGS) $(MACPORTS_CFLAGS) $(READLINE_CFLAGS) $(MSPDEBUG_CFLAGS) -o $@ -c $*.c
