--- mspdebug-0.7-dist/rf2500.c	2010-04-30 14:17:11.000000000 +0900
+++ mspdebug-0.7/rf2500.c	2010-05-11 22:19:41.000000000 +0900
@@ -65,9 +65,11 @@
 		return -1;
 	}
 
+#ifndef __APPLE__
 	if (usb_detach_kernel_driver_np(tr->handle, tr->int_number) < 0)
 		perror("rf2500: warning: can't "
 			"detach kernel driver");
+#endif
 
 	if (usb_claim_interface(tr->handle, tr->int_number) < 0) {
 		perror("rf2500: can't claim interface");
