--- mspdebug/Makefile.~1~	2010-09-26 17:55:39.000000000 +0900
+++ mspdebug/Makefile	2010-10-16 00:47:00.000000000 +0900
@@ -60,7 +60,7 @@
 	  reader.o vector.o output_util.o expr.o fet_error.o binfile.o \
 	  fet_db.o usbutil.o titext.o srec.o device.o coff.o opdb.o output.o \
 	  cmddb.o stdcmd.o prog.o flash_bsl.o
-	$(CC) $(LDFLAGS) $(MACPORTS_LDFLAGS) -o $@ $^ -lusb $(READLINE_LIBS)
+	$(CC) $(LDFLAGS) $(MACPORTS_LDFLAGS) -o $@ $^ $(READLINE_LIBS)
 
 .c.o:
 	$(CC) $(CFLAGS) $(MACPORTS_CFLAGS) $(READLINE_CFLAGS) $(MSPDEBUG_CFLAGS) -o $@ -c $*.c
