diff -ur mspdebug-0.4/Makefile build-msp430-mspdebug/Makefile
--- mspdebug-0.4/Makefile	2010-01-15 10:29:51.000000000 +0900
+++ build-msp430-mspdebug/Makefile	2010-02-27 15:50:38.000000000 +0900
@@ -27,7 +27,7 @@
 
 mspdebug: main.o fet.o rf2500.o dis.o uif.o ihex.o elf32.o stab.o util.o \
 	  bsl.o sim.o
-	$(CC) $(CFLAGS) -o $@ $^ -lusb
+	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $^ -lusb
 
 .c.o:
 	$(CC) $(CFLAGS) -O1 -Wall -o $@ -c $*.c
diff -ur mspdebug-0.4/rf2500.c build-msp430-mspdebug/rf2500.c
--- mspdebug-0.4/rf2500.c	2010-01-15 10:29:51.000000000 +0900
+++ build-msp430-mspdebug/rf2500.c	2010-02-27 15:48:21.000000000 +0900
@@ -56,9 +56,11 @@
 		return -1;
 	}
 
+#if !defined(__APPLE__)
 	if (usb_detach_kernel_driver_np(usbtr_handle, usbtr_int_number) < 0)
 		perror("rf2500: warning: can't "
 			"detach kernel driver");
+#endif
 
 	if (usb_claim_interface(usbtr_handle, usbtr_int_number) < 0) {
 		perror("rf2500: can't claim interface");
