diff -dur mspdebug-0.8-dist/Makefile mspdebug-0.8/Makefile
--- mspdebug-0.8-dist/Makefile	2010-05-22 14:42:10.000000000 +0900
+++ mspdebug-0.8/Makefile	2010-05-23 14:40:49.000000000 +0900
@@ -36,8 +36,10 @@
 	/bin/rm -f mspdebug
 
 install: mspdebug mspdebug.man
-	$(INSTALL) -D -m 0755 -s mspdebug $(PREFIX)/bin/mspdebug
-	$(INSTALL) -D -m 0644 mspdebug.man $(PREFIX)/share/man/man1/mspdebug.1
+	mkdir -p $(PREFIX)/bin
+	$(INSTALL) -m 0755 -s mspdebug $(PREFIX)/bin/mspdebug
+	mkdir -p  $(PREFIX)/share/man/man1
+	$(INSTALL) -m 0644 mspdebug.man $(PREFIX)/share/man/man1/mspdebug.1
 
 .SUFFIXES: .c .o
 
