--- mspdebug-0.5-dist/rf2500.c	2010-03-20 12:39:52.000000000 +0900
+++ mspdebug-0.5/rf2500.c	2010-03-21 02:53:35.000000000 +0900
@@ -56,9 +56,11 @@ static int usbtr_open_interface(struct u
 		return -1;
 	}
 
+#if !defined(__APPLE__)
 	if (usb_detach_kernel_driver_np(usbtr_handle, usbtr_int_number) < 0)
 		perror("rf2500: warning: can't "
 			"detach kernel driver");
+#endif
 
 	if (usb_claim_interface(usbtr_handle, usbtr_int_number) < 0) {
 		perror("rf2500: can't claim interface");
