--- mspdebug-0.5-dist/elf32.c	2010-03-20 12:39:52.000000000 +0900
+++ mspdebug-0.5/elf32.c	2010-03-21 02:51:49.000000000 +0900
@@ -18,11 +18,17 @@
 
 #include <errno.h>
 #include <string.h>
+#ifdef __APPLE__
+#include <libelf.h>
+#else
 #include <elf.h>
+#endif
 #include "binfile.h"
 #include "stab.h"
 
+#ifndef EM_MSP430
 #define EM_MSP430	0x69
+#endif
 
 static const u_int8_t elf32_id[] = {
 	ELFMAG0, ELFMAG1, ELFMAG2, ELFMAG3, ELFCLASS32
