diff -ur mspdebug-0.5-dist/dis.c mspdebug-0.5/dis.c
--- mspdebug-0.5-dist/dis.c	2010-03-20 12:39:52.000000000 +0900
+++ mspdebug-0.5/dis.c	2010-03-21 02:50:35.000000000 +0900
@@ -443,7 +443,7 @@
 	} ops[] = {
 		/* Single operand */
 		{MSP430_OP_RRC,         "RRC"},
-		{MSP430_OP_RRC,         "SWPB"},
+		{MSP430_OP_SWPB,        "SWPB"},
 		{MSP430_OP_RRA,         "RRA"},
 		{MSP430_OP_SXT,         "SXT"},
 		{MSP430_OP_PUSH,        "PUSH"},
